Mathlib/Geometry/Manifold/ContMDiff/Defs.lean

```diff
@@ -730,54 +747,67 @@ theorem contMDiffWithinAt_iff_contMDiffOn_nhds
     exact hsub.trans inter_subset_left
 
 /-- If a function is `C^m` within a set at a point, for some finite `m`, then it is `C^m` within
-this set on an open set around the basepoint.
--/
+this set on an open set around the basepoint. -/
 theorem ContMDiffWithinAt.contMDiffOn'
-    [SmoothManifoldWithCorners I M] [SmoothManifoldWithCorners I' M']
-    {m : ℕ} (hm : (m : ℕ∞) ≤ n)
+    [IsManifold I n M] [IsManifold I' n M']
+    (hm : m ≤ n) (h' : m = ∞ → n = ω)
     (h : ContMDiffWithinAt I I' n f s x) :
     ∃ u, IsOpen u ∧ x ∈ u ∧ ContMDiffOn I I' m f (insert x s ∩ u) := by
-  rcases contMDiffWithinAt_iff_contMDiffOn_nhds.1 (h.of_le hm) with ⟨t, ht, h't⟩
-  rcases mem_nhdsWithin.1 ht with ⟨u, u_open, xu, hu⟩
-  rw [inter_comm] at hu
-  exact ⟨u, u_open, xu, h't.mono hu⟩
+  have : IsManifold I m M := .of_le hm
+  have : IsManifold I' m M' := .of_le hm
+  match m with
+  | ω =>
```

Comment:
You can combine these arms: write `| (m : ℕ) | ω =>` here, and delete the last arm.

Reply:
Good catch!